modules/dnn/src/layers/detection_output_layer.cpp

```diff
@@ -211,11 +212,154 @@ class DetectionOutputLayerImpl : public DetectionOutputLayer
+                             const cv::String& code_type, const bool variance_encoded_in_target,
+                             const bool clip, std::vector<LabelBBox>& all_decode_bboxes)
+    {
+        UMat outmat = UMat(loc_mat.dims, loc_mat.size, CV_32F);
+        size_t nthreads = loc_mat.total();
+        String kernel_name;
+
+        if (code_type == "CORNER")
+            kernel_name = "DecodeBBoxesCORNER";
+        else if (code_type == "CENTER_SIZE")
+            kernel_name = "DecodeBBoxesCENTER_SIZE";
+        else
+            return false;
+
+        for (int i = 0; i < num; ++i)
+        {
+            ocl::Kernel kernel(kernel_name.c_str(), ocl::dnn::detection_output_oclsrc);
+            kernel.set(0, (int)nthreads);
+            kernel.set(1, ocl::KernelArg::PtrReadOnly(loc_mat));
+            kernel.set(2, ocl::KernelArg::PtrReadOnly(prior_mat));
+            kernel.set(3, (int)variance_encoded_in_target);
+            kernel.set(4, (int)numPriors);
+            kernel.set(5, (int)share_location);
+            kernel.set(6, (int)num_loc_classes);
+            kernel.set(7, (int)background_label_id);
+            kernel.set(8, (int)clip);
+            kernel.set(9, ocl::KernelArg::PtrWriteOnly(outmat));
+
+            if (!kernel.run(1, &nthreads, NULL, false))
+                return false;
+        }
+
+        all_decode_bboxes.clear();
+        all_decode_bboxes.resize(num);
+        const float* decode_data = outmat.getMat(ACCESS_READ).ptr<float>();
+        for (int i = 0; i < num; ++i)
+        {
+            LabelBBox& decode_bboxes = all_decode_bboxes[i];
+            for (int c = 0; c < num_loc_classes; ++c)
+            {
+                int label = share_location ? -1 : c;
+                decode_bboxes[label].resize(numPriors);
+                for (int p = 0; p < numPriors; ++p)
+                {
+                    int startIdx = p * num_loc_classes * 4;
+                    util::NormalizedBBox& bbox = decode_bboxes[label][p];
+                    bbox.xmin = decode_data[startIdx + c * 4];
+                    bbox.ymin = decode_data[startIdx + c * 4 + 1];
+                    bbox.xmax = decode_data[startIdx + c * 4 + 2];
+                    bbox.ymax = decode_data[startIdx + c * 4 + 3];
+                }
+            }
+        }
+        return true;
+    }
+
+    void ocl_GetConfidenceScores(const UMat& inp1, const int num,
+                                 const int numPredsPerClass, const int numClasses,
+                                 std::vector<Mat>& confPreds)
+    {
+        int shape[] = { numClasses, numPredsPerClass };
+        for (int i = 0; i < num; i++)
+            confPreds.push_back(Mat(2, shape, CV_32F));
+
+        UMat umat = inp1.reshape(1, num * numPredsPerClass);
+        for (int i = 0; i < num; ++i)
+        {
+            Range ranges[] = { Range(i * numPredsPerClass, (i + 1) * numPredsPerClass), Range::all() };
+            umat(ranges).t().copyTo(confPreds[i]);
```

Comment:
It is better to use `cv::transpose()` directly.

Reply:
code is updated  to use tranpose() here.